lib/solid/parser/base.ex

```diff
@@ -158,17 +158,29 @@ defmodule Solid.Parser.Base do
         ])
         |> map({:erlang, :binary_to_atom, [:utf8]})
 
+      argument_filter =
+        tag(Argument.argument(), :argument)
+        |> tag(
+          repeat(
+            lookahead_not(choice([operator, string("and"), string("or")]))
+            |> concat(filter)
+          ),
+          :filters
+        )
+
+      defcombinatorp(:__argument_filter__, argument_filter)
```

Comment:
Any reason why you had to use a `defcombinatorp` and not simply `argument_filter` ? I think we should avoid using `parsec` for performance reasons:

https://hexdocs.pm/nimble_parsec/NimbleParsec.html#parsec/2-use-cases

> The reason why the above is not recommended is because each parsec/2 combinator ends-up adding a stacktrace entry during parsing, which affects the ability of NimbleParsec to optimize code. If the goal is to compose combinators, you can do so with modules and functions:

I know that the parser is getting pretty slow because of its size and using `defcombinatorp` should speed it up a bit but I'm also not sure if we want to do this before running some benchmarks and trying other techniques.

The only reason why I had to implement `liquid_entry` with `parsec` was so that I could have an indirect recursion where a `liquid_entry` could have nested `liquid_entry`s.

Reply:
Thanks for your comment, I will do some test to see if there is any better solution

Reply:
I think if you use `argument_filter` directly instead of `parsec(__...` it  should work 🤔